patroni/psycopg.py

```diff
@@ -14,27 +22,80 @@
     except ImportError:
         _legacy = True
 
-    def quote_literal(value, conn=None):
+    if TYPE_CHECKING:
+        from psycopg2.extensions import connection
+
+    def quote_literal(value: Any, conn: Optional["connection"] = None) -> str:
```

Comment:
Actually... the difference between psycopg2 connection and psycopg3 Connection makes pyright crazy...
Hence I suggest the following:
```suggestion
    def quote_literal(value: Any, conn: Optional[Any] = None) -> str:
```